src/lexer.ts

```diff
@@ -317,7 +337,7 @@ export class JsonTemplateLexer {
   }
 
   private static isIdStart(ch: string) {
-    return ch === '$' || ch === '_' || (ch >= 'a' && ch <= 'z') || (ch >= 'A' && ch <= 'Z');
```

Comment:
Not supporting `$` wouldn't be a breaking change for cdkv2 integrations ?

Reply:
I am supporting $, just I have change how I was handling it, all existing tests cases are passing so no need to worry